moveit_core/robot_model/src/joint_model_group.cpp

```diff
@@ -113,154 +126,153 @@ JointModelGroup::JointModelGroup(const std::string& group_name, const srdf::Mode
+
+  joint_variables_index_map_ =
+      views::concat(
+          views::zip(not_fixed_joints_view | views::transform(get_variable_names_view), not_fixed_start_index_view) |
+              views::transform([](const auto& e) {
+                const auto& [names, start_index] = e;
+                return views::zip(names, views::iota(start_index, start_index + names.size()));
+              }) |
+              views::join |
+              views::transform([](const auto& e) { return std::make_pair(std::get<0>(e), std::get<1>(e)); }),
+          views::zip(not_fixed_joints_view | views::transform(get_name), not_fixed_start_index_view) |
+              views::transform([](const auto& e) { return std::make_pair(std::get<0>(e), std::get<1>(e)); })) |
+      to<std::map>();
+
+  continuous_joint_model_vector_ =
+      not_fixed_joints_view | views::filter([](const auto& jm) {
+        return jm->getType() == JointModel::REVOLUTE && static_cast<const RevoluteJointModel*>(jm)->isContinuous();
+      }) |
+      to<std::vector>();
+
+  variable_count_ = accumulate(not_fixed_joints_view | views::transform(get_variable_count), 0);
 
   // now we need to find all the set of joints within this group
   // that root distinct subtrees
-  for (const JointModel* active_joint_model : active_joint_model_vector_)
-  {
-    // if we find that an ancestor is also in the group, then the joint is not a root
-    if (!includesParent(active_joint_model, this))
-      joint_roots_.push_back(active_joint_model);
-  }
+  joint_roots_ = active_joint_model_vector_ |
+                 views::filter([this](const auto& jm) { return !includesParent(jm, this); }) | to<std::vector>();
 
   // when updating this group within a state, it is useful to know
   // if the full state of a group is contiguous within the full state of the robot
-  if (variable_index_list_.empty())
-    is_contiguous_index_list_ = false;
-  else
-    for (std::size_t i = 1; i < variable_index_list_.size(); ++i)
-      if (variable_index_list_[i] != variable_index_list_[i - 1] + 1)
-      {
-        is_contiguous_index_list_ = false;
-        break;
-      }
-
-  // when updating/sampling a group state only, only mimic joints that have their parent within the group get updated.
-  for (const JointModel* mimic_joint : mimic_joints_)
-    // if the joint we mimic is also in this group, we will need to do updates when sampling
-    if (hasJointModel(mimic_joint->getMimic()->getName()))
-    {
-      int src = joint_variables_index_map_[mimic_joint->getMimic()->getName()];
-      int dest = joint_variables_index_map_[mimic_joint->getName()];
-      GroupMimicUpdate mu(src, dest, mimic_joint->getMimicFactor(), mimic_joint->getMimicOffset());
-      group_mimic_update_.push_back(mu);
-    }
+  is_contiguous_index_list_ =
+      !variable_index_list_.empty() && none_of(variable_index_list_ | views::sliding(2), [](const auto& sub_view) {
+        const auto& values = sub_view | to<std::vector>();
```

Comment:
Feels like there should be a way around this type conversion

Reply:
As far as I can tell this is the most straightforward way to be able to index the elements of a view.  I agree it is clunky and I feel like there should be something better.